cli.spec.js

```diff
@@ -0,0 +1,79 @@
+const test = require('ava');
+const fs = require('fs');
+const mkdirp = require('mkdirp');
+const rimraf = require('rimraf');
+const { exec } = require("child_process");
+
+
+test.before(t => {
+  /* Lets create this structure of folders and files
+    /test  
+      /first
+        /fourth
+          F.js
+          F.spec.js
+        A.js
+        B.js
+        C.js
+        C.spec.js
+      /second
+        D.js
+        E.js
+      /third
+        TEST.spec.js
+  */
+  const folders = [
+    './test/first/fourth',
+    './test/second',
+    './test/third',
+  ];
+  const files = [
+    './test/first/A.js',
+    './test/first/B.js',
+    './test/first/C.js',
+    './test/first/C.spec.js',
+    './test/first/fourth/F.js',
+    './test/first/fourth/F.spec.js',
+    './test/second/D.js',
+    './test/second/E.js',
+    './test/third/TEST.spec.js',
+  ];
+  return Promise.all(
+    folders.map((folder) => mkdirp(folder).catch((err) => t.log(err)))
+  ).then(() => {
+    files.forEach((file) => fs.writeFile(file, '', () => {}));
+  });
+});
+
+test.after.always('Guaranteed cleanup', () => {
+  return new Promise((resolve, reject) => {
+    rimraf('{test,first,second}', (err) => err ? reject(error) : resolve());
+  });
+});
+
+test.serial('should link the directories and the files which match the pattern', (t) => {
+  return new Promise((resolve, reject) => {
```

Comment:
This kind of test are okey to check if the feedback produced is the expected, but this is not a real testing. The test should check the generated files.